CentralComputing/Sensor_Test.cpp

```diff
@@ -0,0 +1,388 @@
+		// ignored
+	}
+	// error catching
+	else {
+		// error occured in reading commands
+		cout << "ERROR: instruction (" << word << ") undefined" << endl;
+		cout << "       Result: Ignored" << endl;
+	}
+	return true;
+}
+
+/**
+	Reads from a file and executes each command one by one
+	fileName: file to read in
+	pod: reference to pod object to be executed on
+	
+	prints at the end if the test was succesful
+*/
+void readFile( string fileName , Pod& pod) {
+	bool worked = true;
+	ifstream wordsFile(fileName);
+	string word;
+	if(wordsFile.is_open()) {
+		// reads file line by line
+		while (getline(wordsFile, word)) {
+			//cout << "~" << word << endl;
+			worked &= process(word, pod);
+		}
+	}
+
+	cout << "Test was successful = " << worked << endl;
+}
+
+/**
+	Reads from command line and executes user commands
+	pod: reference to pod object to be executed on
+*/
+void manual(Pod& pod) {
+	string line = "";
+	// loop until user types end or quit
+	while(true) {
+		// prints the carrot for beauty's sake
+		cout << ">";
+		
+		// reads from command line
+		cin >> line;
+
+		// a few special instructions that process func doesn't include
+		if(line == "help" || line == "-h" ) {
+			printHelp();
+		}
+		if(line == "help_instructions" || line == "-hi" ) {
+			printHelpInstr();
+		}
+		// exit condition for loop
+		else if(line == "end" || line == "exit" ) {
+			return;
+		}
+		else {
+			// runs through the same process func that readFile uses
+			process(line, pod);
+		}
+	}
+}
+
+/**
+	Monkey test, does random functional calls to see if pod would break
+	num: the number of commands to run before terminating
+	pod: reference to pod object to be executed on
+*/
+void random(int num, Pod& pod) {
+	// sets seed for rng
+	srand(time(NULL));
+
+	cout << "random test for " << num << "cycles" << endl;
+	
+	for(int i = 0; i < num; i++) {
+		// change the m in (rand() % m + 1) to change the maximum value
+		int r = rand() % 6 +1;
+		// each possible int has a correlated function
+		if ( r == 1 ) {
+			cout << "~move_functinal_tests"<< endl;
+			pod.move_functional_tests();
+		}
+		else if ( r == 2 ) {
+			cout << "~move_safe_mode"<< endl;
+			pod.move_safe_mode();
+		}
+		else if ( r == 3 ) {
+			cout << "~move_flight_acceleration"<< endl;
+			pod.move_flight_acceleration();
+		}
+		else if ( r == 4 ) {
+			cout << "~emergency_brake"<< endl;
+			pod.emergency_brake();
+		}
+		else if ( r == 5 ) {
+			cout << "~coast"<< endl;
+			pod.coast();
+		}
+		else if ( r == 6 ) {
+			cout << "~brake"<< endl;
+			pod.brake();
+		}
+	}
+}
+
+/**
+	A default test for the pod, for easy testing
+	pod: reference to pod object to be executed on
+*/
+void defaultTest(Pod& pod) {
+	cout << "doing default test here" << endl;
+	//pod.move_functional_tests();
+	cout << pod.get_current_state_string() << endl;
+}
+/**
+	main function
+	param: an array of strings to be parsed into a command";
+*/
+int main(int argc, char** argv) {
```

Comment:
For now, this is fine as main. However, this will for sure be changed in the future